MoreLinq/Experimental/Batch.cs

```diff
@@ -268,9 +268,10 @@ public bool UpdateWithNext()
 
             public override T this[int index]
             {
+#pragma warning disable CA2201 // Do not raise reserved exception types
                 get => index >= 0 && index < Count ? _array[index] : throw new IndexOutOfRangeException();
```

Comment:
Per https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.item?view=net-7.0, should be `ArgumentOutOfRangeException` anyway. Possibly breaking change, but highly unlikely anyone is specifically catching for `IndexOutOfRangeException`.

Reply:
Too bad that this didn't get caught during the review of PR #856, but it's not too late to change since this overload of `Batch` hasn't been released yet.

---
PS The summary of [`IndexOutOfRangeException`](https://learn.microsoft.com/en-us/dotnet/api/system.indexoutofrangeexception) really needs to be changed to avoid mentioning _collection_:

> The exception that is thrown when an attempt is made to access an element of an array or collection with an index that is outside its bounds.

The examples in the documentation don't help much with mixed uses:

> Typically, an [IndexOutOfRangeException](https://learn.microsoft.com/en-us/dotnet/api/system.indexoutofrangeexception?view=net-7.0) exception is thrown as a result of developer error. Instead of handling the exception, you should diagnose the cause of the error and correct your code. The most common causes of the error are:

>
> - Forgetting that the upper bound of a collection or a zero-based array is one less than its number of members or elements, as the following example illustrates.

>
>   ```c#
>   using System;
>   using System.Collections.Generic;
>   
>   public class Example
>   {
>      public static void Main()
>      {
>         List<Char> characters = new List<Char>();
>         characters.InsertRange(0, new Char[] { 'a', 'b', 'c', 'd', 'e', 'f' } );
>         for (int ctr = 0; ctr <= characters.Count; ctr++)
>            Console.Write("'{0}'    ", characters[ctr]);
>      }
>   }
>   // The example displays the following output:
>   //    'a'    'b'    'c'    'd'    'e'    'f'
>   //    Unhandled Exception:
>   //    System.ArgumentOutOfRangeException:
>   //    Index was out of range. Must be non-negative and less than the size of the collection.
>   //    Parameter name: index
>   //       at Example.Main()
>   ```

Reply:
Follow-up in PR #927.